modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/configuration/CETConfigurationFactory.java

```diff
@@ -74,25 +108,112 @@ protected void modified(Map<String, Object> properties) throws Exception {
 				_cet = _cetManager.addCET(
 					cetConfiguration, companyId,
 					_getExternalReferenceCode(properties));
+
+				if (Objects.equals(
```

Comment:
I'm just wanting to make sure that lifecycle works correctly in this case. This might be totally okay since we only have workspace client extensions for this.

But, does the `deactivate` method get called when you shutdown the server for instance.

Reply:
@ethib137  Yes, the deactivate method is called whenever the client extension is deployed or the serve turns down.